### ISSUE TITLE
The camouflage parameter is not connected. 

### ISSUE THREAD

**Issue by combx** [aminvakil/docker-ocserv#75]
What I do: 
_docker pull quay.io/aminvakil/ocserv_

_docker run --name ocserv --sysctl net.ipv4.ip_forward=1 --cap-add NET_ADMIN --security-opt no-new-privileges -p 443:443 -p 443:443/udp -v /etc/letsencrypt/live/xxx.xx/privkey.pem:/etc/ocserv/certs/server-key.pem -v /etc/letsencrypt/live/xxx.xx/fullchain.pem:/etc/ocserv/certs/server-cert.pem -d quay.io/aminvakil/ocserv_

_docker exec ocserv sed -i '/^camouflage = /{s/false/true/}' /etc/ocserv/ocserv.conf_
_docker exec ocserv sed -i '/^camouflage_secret = /{s/mysecretkey/xxxsecret/}' /etc/ocserv/ocserv.conf_
_docker exec -ti ocserv ocpasswd -c /etc/ocserv/ocpasswd xxx_

At the moment the connection to the server is working, but to any URL. Camouflage function is not active

After
_docker restart ocserv_
settings regarding Camouflage are reset to factory settings (false, mysecretkey)

Why doesn't the Camouflage option work?

**Comment by aminvakil** [comment]
This is happening after https://github.com/aminvakil/docker-ocserv/pull/72 got merged and `ocserv.conf` changes will not persist after a restart which is fine IMO.

You should persist your `ocserv.conf` and mount it into your container, make changes on your host and then restart your container.

Or you can make `camouflage` settings an environment variable in a PR, I'll be happy to review and merge it.

**Comment by combx** [comment]
_This is happening after https://github.com/aminvakil/docker-ocserv/pull/72 got merged and ocserv.conf changes will not persist after a restart which is fine IMO._
Got it

_You should persist your ocserv.conf and mount it into your container, make changes on your host and then restart your container._
Agree. This is the first thing I did.
But I got an interesting result.
I add this code to the container launch line
**-v /root/aminvakil/ocserv.conf:/etc/ocserv/ocserv.conf -v /root/aminvakil/ocpasswd:/etc/ocserv/ocpasswd**

And after
**docker restart ocserv**
the ocserv.conf config file is rewritten to the default one, and the ocpasswd file remains with the data.

If I run the add new user command I get an error
**Cannot write to '/etc/ocserv/ocpasswd'.**
If I run the command to change parameters in ocserv.conf, the following error appears
**sed: can't move '/etc/ocserv/ocserv.confjbiJNe' to '/etc/ocserv/ocserv.conf': Resource busy**
Perhaps this is related to rights and the container does not have the rights to write to /root/aminvakil/ocserv.conf

And I have some problems with the network. After connecting to the VPN from my phone, there is no Internet access

**Comment by aminvakil** [comment]
Yes, my configurations have got lost as well, I've reverted #72, please pull the latest image.

**Comment by combx** [comment]
> Yes, my configurations have got lost as well, I've reverted #72, please pull the latest image.

Failed to launch(
I tried to run it in two ways
first way
git pull origin master
docker build .
and after launching the image
docker run --name ocserv --sysctl net.ipv4.ip_forward=1 --cap-add NET_ADMIN --security-opt no-new-privileges -p 443:443 -p 443:443/udp -d quay.io/ aminvakil/ocserv
connection via test user occurs, but the Internet does not work

second way
docker pull quay.io/aminvakil/ocserv
docker-compose up -d
connection occurs, but the Internet does not work

maybe I'm doing something wrong, I'm sad

**Comment by aminvakil** [comment]
> maybe I'm doing something wrong, I'm sad

Try with exact tag:
```
docker run --name ocserv --sysctl net.ipv4.ip_forward=1 --cap-add NET_ADMIN --security-opt no-new-privileges -p 443:443 -p 443:443/udp -d quay.io/aminvakil/ocserv:1.3.0-5
```

**Comment by combx** [comment]
> > maybe I'm doing something wrong, I'm sad
> 
> Try with exact tag:
> 
> ```
> docker run --name ocserv --sysctl net.ipv4.ip_forward=1 --cap-add NET_ADMIN --security-opt no-new-privileges -p 443:443 -p 443:443/udp -d quay.io/aminvakil/ocserv:1.3.0-5
> ```

I tried different options.
I connected my config file and passwords.
launched with standard parameters.
does not work(

last log with docker-compose

ocserv  | Signing certificate...
ocserv  | Create test user 'test' with password 'test'
ocserv  | sysctl: error setting key 'net.ipv4.ip_forward': Read-only file system
ocserv  | note: vhost:default: setting 'plain' as primary authentication method
ocserv  | note: the cisco-client-compat option implies select-group-by-url = true; enabling
ocserv  | note: setting 'file' as supplemental config option
ocserv  | listening (TCP) on 0.0.0.0:443...
ocserv  | listening (TCP) on [::]:443...
ocserv  | listening (UDP) on 0.0.0.0:443...
ocserv  | listening (UDP) on [::]:443...

last log with docker

[@76178 ocserv1]# docker logs --tail=50 38e0ad0a57e4
sysctl: error setting key 'net.ipv4.ip_forward': Read-only file system
error: cannot access config file: /etc/ocserv/ocserv.conf
Usage: ocserv -c [config]
Use ocserv --help for more information.
